kfs-core/src/main/java/edu/arizona/kfs/tax/document/IncomeTypeHandler.java

```diff
@@ -0,0 +1,579 @@
+					retval = true;
+					break;
+				}
+			}
+		}
+		return retval;
+	}
+
+	/**
+	 * This method validates the DocumentIncomeType entry when the user hits the "Add" button on the screen
+	 * 
+	 * @return
+	 */
+	public boolean validateIncomeType() {
+		boolean retval = true;
+
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("validateIncomeType()");
+		}
+
+		MessageMap errors = GlobalVariables.getMessageMap();
+
+		// make sure chart is selected
+		if (StringUtils.isBlank(newIncomeType.getChartOfAccountsCode())) {
+			errors.putError("newIncomeTypeError1", TaxKeyConstants.ERROR_INCOME_TYPE_COA_REQUIRED);
+			retval = false;
+		}
+
+		// make sure income type is selected
+		if (StringUtils.isBlank(newIncomeType.getIncomeTypeCode())) {
+			errors.putError("newIncomeTypeError2", TaxKeyConstants.ERROR_INCOME_TYPE_CODE_REQUIRED);
+			retval = false;
+		}
+
+		// make sure we have an amount entered
+		if (newIncomeType.getAmount() == null) {
+			errors.putError("newIncomeTypeError3", TaxKeyConstants.ERROR_INCOME_TYPE_AMOUNT_REQUIRED);
+			retval = false;
+		}
+
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("validateIncomeType() retval: " + retval);
+		}
+
+		return retval;
+	}
+
+	public CurrencyFormatter getCurrencyFormatter() {
+		return currencyFormatter;
+	}
+
+	public T1 getNewIncomeType() {
+		if (newIncomeType == null) {
+			newIncomeType = getInitializedDocumentIncomeType();
+		}
+
+		return newIncomeType;
+	}
+
+	private T1 getInitializedDocumentIncomeType() {
+		T1 retval = null;
+
+		try {
+			// use the class passed in the constructor to create a new instance
+			retval = incomeTypeClass.newInstance();
+
+			// initialize amount to 0
+			retval.setAmount(new KualiDecimal(0));
+
+			// set the document identifier from the document
+			retval.setDocumentIdentifier(incomeTypeDocumentContainer.getDocumentIdentifier());
+		}
+
+		catch (Exception ex) {
+			LOG.error(ex);
+		}
+
+		return retval;
+	}
+
+	public void setNewIncomeType(T1 newIncomeType) {
+		this.newIncomeType = newIncomeType;
+	}
+
+	/**
+	 * This method loads IncomeTypes into a map keyed by box number
+	 * 
+	 * @return
+	 */
+	private Map<String, IncomeType> getIncomeTypeMap() {
+		Map<String, IncomeType> retval = new HashMap<String, IncomeType>();
+
+		// Load the IncomeTypes from the database
+		List<IncomeType> incomeTypes = (List<IncomeType>) getBusinessObjectService().findAll(IncomeType.class);
+
+		if (incomeTypes != null) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("found " + incomeTypes.size() + " income types");
+			}
+
+			// create IncomeType map by 1099 box
+			for (IncomeType ic : incomeTypes) {
+				if (StringUtils.isNotBlank(ic.getIncomeTypeBox())) {
+					if (LOG.isDebugEnabled()) {
+						LOG.debug("IncomeTypeCode: " + ic.getIncomeTypeCode() + ", box: " + ic.getIncomeTypeBox());
+					}
+
+					retval.put(ic.getIncomeTypeBox(), ic);
+				}
+			}
+		}
+
+		return retval;
+	}
+

```

Comment:
The Use Tax functionality was removed for 2 reasons: 
1) It was not listed in specs. 
2) It requires functionality that is in purap, and this file is required by DV in core.

Should I put this back in, or wait until the kfs-tax refactor?